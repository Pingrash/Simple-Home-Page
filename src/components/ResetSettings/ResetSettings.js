import React from 'react';
import PropTypes from 'prop-types';

import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  spacing: {
    margin: '1rem 0'
  }
}));

const ResetSettings = ({ updateUser, resetLinks }) => {
  const classes = useStyles();
  return (
    <>
      <Typography component='h4' variant='h4'>
        Reset To Default
      </Typography>
      <Typography
        component='span'
        variant='body1'
        color='error'
        data-test='warning'
        className={classes.spacing}
      >
        WARNING: Pressing this button will reset everything to
        default!
      </Typography>
      <Button
        color='secondary'
        variant='outlined'
        onClick={e => {
          e.preventDefault();
          updateUser('');
          resetLinks();
        }}
        className={classes.spacing}
        data-test='reset-button'
      >
        Delete User
      </Button>
    </>
  );
};

ResetSettings.propTypes = {
  updateUser: PropTypes.func.isRequired,
  resetLinks: PropTypes.func.isRequired
};

export default ResetSettings;
