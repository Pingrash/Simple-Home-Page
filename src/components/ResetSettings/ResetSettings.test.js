import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {
  findByDataTest,
  checkProps,
  testFunc
} from '../../../Utilities';
import ResetSettings from './ResetSettings';

const props = {
  updateUser: testFunc,
  resetLinks: testFunc
};

const setup = () => {
  const component = shallow(<ResetSettings {...props} />);
  return component;
};

describe('Snapshot', () => {
  const tree = renderer.create(<ResetSettings {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('ResetSettings component renders correctly', () => {
  let wrapper;
  beforeEach(() => (wrapper = setup()));

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render a warning message', () => {
    const message = findByDataTest(wrapper, 'warning');
    expect(message.length).toBe(1);
  });

  it('Should render a delete button', () => {
    const button = findByDataTest(wrapper, 'reset-button');
    expect(button.length).toBe(1);
  });
});

describe('ResetSettings should have correct props', () => {
  it('Should not return a props error', () => {
    const propsError = checkProps(ResetSettings, props);
    expect(propsError).toBeUndefined();
  });
});
