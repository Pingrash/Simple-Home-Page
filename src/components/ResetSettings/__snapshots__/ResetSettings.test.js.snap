// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <h4
    className="MuiTypography-root MuiTypography-h4"
  >
    Reset To Default
  </h4>,
  <span
    className="MuiTypography-root makeStyles-spacing-1 MuiTypography-body1 MuiTypography-colorError"
    data-test="warning"
  >
    WARNING: Pressing this button will reset everything to default!
  </span>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-spacing-1 MuiButton-outlinedSecondary"
    data-test="reset-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Delete User
    </span>
  </button>,
]
`;
